crypto/inc/mbedtls_config_default.h

```diff
@@ -21,10 +21,19 @@
  *  This file is part of mbed TLS (https://tls.mbed.org)
  */
 
-#if PLATFORM_ID == 6 || PLATFORM_ID == 8
+#include "hal_platform.h"
+
+#if PLATFORM_ID==6 || PLATFORM_ID==8
+#define PLATFORM_WICED 1
```

Comment:
`HAL_PLATFORM_WICED` in `hal_platform.h` perhaps?

We should also get rid of these `mbedlts_config_default.h`/`mbedtls_config_photon.h` in the future and use the same scheme for Gen 2 devices as for Gen 3: platform-specific `mbedtls_config_platform.h` in platform-specific HAL folder.